src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -526,18 +527,13 @@ public static string Base64Decode(string base64EncodedData)
         /// <returns>The string representing nextLink.</returns>
         public static JsonElement CreateNextLink(string path, NameValueCollection? nvc, string after)
         {
-            if (nvc is null)
-            {
-                nvc = new();
-            }
-
+            string queryString = FormatQueryString(queryParameters: nvc);
             if (!string.IsNullOrWhiteSpace(after))
             {
-                nvc["$after"] = after;
+                string afterPrefix = string.IsNullOrWhiteSpace(queryString) ? "?" : "&";
+                queryString += $"{afterPrefix}{RequestParser.AFTER_URL}={after}";
             }
```

Comment:
You could take advantage of the .NET helper inside our `FormatQueryString` function: `QueryHelpers.AddQueryString()`, to avoid creating your own conditional logic determining whether to use the `?` or `&` prefix.

Also, i see this logic ensures we no longer add a hard coded query param key name: `nvc["$after"]`. Was there a bug you identified that required updating the code/could a test be added to address that?

Reply:
> You could take advantage of the .NET helper inside our `FormatQueryString` function: `QueryHelpers.AddQueryString()`, to avoid creating your own conditional logic determining whether to use the `?` or `&` prefix.



That helper function is what causes the incorrect URI encoding within the after, which is why it was separated out to how we have it now.


> Also, i see this logic ensures we no longer add a hard coded query param key name: `nvc["$after"]`. Was there a bug you identified that required updating the code/could a test be added to address that?

It is because we had these constants with the query param names as private inside of the request parser, but now we are using them in other parts of the code, so they are public. So now I think we should just be re-using them whenever possible so that everything has 1 source of truth for those names. Ultimately it might make sense to not have just a request parser class, since it is a bit weird to have the "parser" holding names like this, but rather have a request class that has these things and then also parsing methods or sub classes etc

for now this makes the most sense though I think.

Reply:
thanks for the clarification!

Reply:
Optional, but would be helpful for future devs:
1. A comment above the following line to warn of the side effects that you learned about when making this fix:
```csharp
queryString = QueryHelpers.AddQueryString(queryString, key, queryParamValues);
```
2. Comment for `FormatQueryString(...)` warning of the side effects